### PR TITLE
hotplug_block_reszie: New cases to test hot plug block device then resize

### PR DESCRIPTION
**Pull request by YongxueHong**
New 8 cases:
1. Hot plug a block device then enlarge it, shutdown guest by qmp.
2. Hot plug a block device then enlarge it, shutdown guest by shel.
3. Hot plug a block device then enlarge it, reboot guest by qmp.
4. Hot plug a block device then enlarge it, reboot guest by shell.
5. Hot plug a block device then shrink it, shutdown guest by qmp.
6. Hot plug a block device then shrink it, shutdown guest by shell.
7. Hot plug a block device then shrink it, reboot guest by qmp.
8. Hot plug a block device then shrink it, reboot guest by shell.

ID: 1703910
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on #1735

File: qemu/tests/hotplug_block_resize.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+from virttest import error_context
+from virttest import utils_disk
+from virttest import utils_numeric
+from virttest import utils_test
+from virttest import storage
+from virttest import qemu_storage
+from virttest import data_dir
+
+from provider.plug_block_devices import SerialPlugManger
+from provider.storage_benchmark import generate_instance
```
Comment (kimi1978wy) [inline qemu/tests/hotplug_block_resize.py]: There is no `provider.plug_block_devices` for `provider/block_devices_plug.py`, so it occurred error as` ERROR: No module named 'provider.plug_block_devices'`
And a typo for SerialPlugManger, it should be "Manager"

**Comment by kimi1978wy** [comment]
It failed when test shutdown_vm cases:
It still try to login guest after shutdown vm.

13:53:12 ERROR| Traceback (most recent call last):
13:53:12 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-72.0-py3.6.egg/virttest/error_context.py", line 135, in new_fn
13:53:12 ERROR|     return fn(*args, **kwargs)
13:53:12 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_block_resize.py", line 113, in run
13:53:12 ERROR|     iozone.clean()
13:53:12 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/provider/storage_benchmark.py", line 256, in clean
13:53:12 ERROR|     session = self.vm.wait_for_login(timeout=360)
13:53:12 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-72.0-py3.6.egg/virttest/virt_vm.py", line 1177, in wait_for_login
13:53:12 ERROR|     self.verify_alive()
13:53:12 ERROR|   File "/usr/local/lib/python3.6/site-packages/avocado_framework_plugins_vt-72.0-py3.6.egg/virttest/qemu_vm.py", line 227, in verify_alive
13:53:12 ERROR|     self.process.get_output())
13:53:12 ERROR| virttest.virt_vm.VMDeadError: VM is dead

**Comment by zhenyzha** [comment]
Test Results:
 (1/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm: PASS (126.19 s)
 (2/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm: PASS (192.75 s)
 (3/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm: PASS (139.03 s)
 (4/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm: PASS (225.33 s)
 (5/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm: PASS (133.89 s)
 (6/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm: PASS (197.95 s)
 (7/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm: PASS (122.23 s)
 (8/8) Host_RHEL.m8.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm: PASS (537.28 s)
RESULTS    : PASS 8 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1677.80 s
JOB HTML   : /home/kar/workspace/job-results/job-2019-11-17T21.26-b38e3d2/results.html

 (01/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_d
isk.send_qmp.shutdown_vm: PASS (120.49 s)
 (02/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_d
isk.send_qmp.shutdown_vm.with_scsi_hba: PASS (146.49 s)
 (03/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_d
isk.send_qmp.reboot_vm: PASS (201.92 s)
 (04/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (210.34 s)
 (05/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm: PASS (120.09 s)
 (06/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (134.07 s)
 (07/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm: PASS (229.37 s)
 (08/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm.with_scsi_hba: PASS (249.04 s)
 (09/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm: PASS (119.05 s)
 (10/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm.with_scsi_hba: PASS (143.05 s)
 (11/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm: PASS (227.12 s)
 (12/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (245.80 s)
 (13/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm: PASS (140.94 s)
 (14/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (117.63 s)
 (15/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm: PASS (262.75 s)
 (16/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.ppc64le.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm.with_scsi_hba: PASS (222.72 s)
RESULTS    : PASS 16 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 2895.53 s
JOB HTML   : /home/kar/workspace/job-results/job-2019-11-18T02.06-087c907/results.html

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Comment by peixiu** [comment]
(01/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (4205.74 s)
 (02/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm: PASS (162.22 s)
 (03/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm.with_scsi_hba: PASS (145.33 s)
 (04/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm: PASS (197.25 s)
 (05/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (188.26 s)
 (06/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm: PASS (133.73 s)
 (07/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (123.81 s)
 (08/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm: PASS (84.82 s)
 (09/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm.with_scsi_hba: PASS (222.19 s)
 (10/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm: PASS (121.87 s)
 (11/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm.with_scsi_hba: PASS (137.27 s)
 (12/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm: PASS (180.57 s)
 (13/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (181.26 s)
 (14/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm: PASS (122.51 s)
 (15/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (138.38 s)
 (16/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm: PASS (212.00 s)
 (17/17) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm.with_scsi_hba: PASS (81.71 s)

**Comment by peixiu** [comment]
LGTM~
Acked-By: Peixiu Hou phou@redhat.com

File: qemu/tests/cfg/hotplug_block_resize.cfg [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    virtio_blk:
+        driver_name = viostor
+    virtio_scsi:
+        driver_name = vioscsi
+    Windows:
+        iozone_options = '-s 4k -s 64k -s 1m -b C:\{0}_stress_test -g {1} -I -i 0 -i 1 -f {0}:\iozone_test'
+    Linux:
+        iozone_options = '-s 4k -s 64k -s 1m -g {1} -I -i 0 -i 1 -f {0}/iozone_test'
+    variants:
+        - @default:
```
Comment (qingwangrh) [inline qemu/tests/cfg/hotplug_block_resize.cfg]: IMHO, do not hide variants, it is easier to address the test

Comment (YongxueHong) [reply qemu/tests/cfg/hotplug_block_resize.cfg]: Okay, I have updated it, thanks.

**Comment by kimi1978wy** [comment]
LGTM
Acked-By: Yu Wang <wyu@redhat.com>

(1/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm: PASS (49.62 s) (2/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm: PASS (70.31 s) (3/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm: PASS (52.19 s) (4/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm: PASS (79.92 s) (5/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm: PASS (50.02 s) (6/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm: PASS (71.83 s) (7/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm: PASS (52.69 s) (8/8) Host_RHEL.m8.u1.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm: PASS (82.67 s)

**Comment by qingwangrh** [comment]
./workspace/bin/python3 ConfigTest.py --testcase=hotplug_block_resize --guestname=8.1.1 --clone=no

(01/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm: PASS (57.72 s)
 (02/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.shutdown_vm.with_scsi_hba: PASS (56.67 s)
 (03/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm: PASS (82.94 s)
 (04/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (80.45 s)
 (05/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm: PASS (58.78 s)
 (06/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (56.57 s)
 (07/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm: PASS (90.98 s)
 (08/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.enlarge_disk.send_shell.reboot_vm.with_scsi_hba: PASS (88.11 s)
 (09/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm: PASS (57.74 s)
 (10/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.shutdown_vm.with_scsi_hba: PASS (58.42 s)
 (11/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm: PASS (86.38 s)
 (12/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_qmp.reboot_vm.with_scsi_hba: PASS (83.32 s)
 (13/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm: PASS (59.72 s)
 (14/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.shutdown_vm.with_scsi_hba: PASS (60.12 s)
 (15/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm: PASS (91.35 s)
 (16/16) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.hotplug_block_resize.shrink_disk.send_shell.reboot_vm.with_scsi_hba: PASS (91.50 s)

LGTM.

File: qemu/tests/cfg/hotplug_block_resize.cfg [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    virtio_blk:
+        driver_name = viostor
+    virtio_scsi:
+        driver_name = vioscsi
+    Windows:
+        iozone_options = '-s 4k -s 64k -s 1m -b C:\{0}_stress_test -g {1} -I -i 0 -i 1 -f {0}:\iozone_test'
+    Linux:
+        iozone_options = '-s 4k -s 64k -s 1m -g {1} -I -i 0 -i 1 -f {0}/iozone_test'
+    variants:
+        - without_scsi_hba:
```
Comment (peixiu) [inline qemu/tests/cfg/hotplug_block_resize.cfg]: better consider blk disk hotplug situation~  it also will hotplug blk hba when hotplug a disk.

Comment (YongxueHong) [reply qemu/tests/cfg/hotplug_block_resize.cfg]: Talked with @qingwangrh , persist in the previous variant` @default`.

**Comment by YongxueHong** [comment]
Thanks all, let's merge it.

**Pull request merged**